From 768ec33e9ee560580c85e367c8abca31de7d7198 Mon Sep 17 00:00:00 2001
From: Fangrui Song <i@maskray.me>
Date: Mon, 16 Dec 2019 22:11:31 -0800
Subject: [PATCH 2/2] Adapt llvmorg-10-init-12036-g3b9715cb2193:
 handleDeclOccurence -> handleDeclOccurrence

(cherry picked from commit 86e340caf00b2d337ddb839e646085bf2222be3b)
---
 src/indexer.cc | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/indexer.cc b/src/indexer.cc
index 0459103f..33481161 100644
--- a/src/indexer.cc
+++ b/src/indexer.cc
@@ -677,10 +677,13 @@ public:
 public:
   IndexDataConsumer(IndexParam &param) : param(param) {}
   void initialize(ASTContext &ctx) override { this->ctx = param.ctx = &ctx; }
-  bool handleDeclOccurence(const Decl *d, index::SymbolRoleSet roles,
-                           ArrayRef<index::SymbolRelation> relations,
-                           SourceLocation src_loc,
-                           ASTNodeInfo ast_node) override {
+#if LLVM_VERSION_MAJOR < 10 // llvmorg-10-init-12036-g3b9715cb219
+# define handleDeclOccurrence handleDeclOccurence
+#endif
+  bool handleDeclOccurrence(const Decl *d, index::SymbolRoleSet roles,
+                            ArrayRef<index::SymbolRelation> relations,
+                            SourceLocation src_loc,
+                            ASTNodeInfo ast_node) override {
     if (!param.no_linkage) {
       if (auto *nd = dyn_cast<NamedDecl>(d); nd && nd->hasLinkage())
         ;
-- 
2.26.0

